ringpop.go

```diff
@@ -39,78 +39,37 @@ package ringpop
 
 import (
 	"bytes"
+	"errors"
 	"fmt"
-	"io/ioutil"
 	"sync"
 	"time"
 
-	"github.com/Sirupsen/logrus"
 	"github.com/apache/thrift/lib/go/thrift"
 	"github.com/dgryski/go-farm"
 	log "github.com/uber-common/bark"
 	"github.com/uber/ringpop-go/forward"
+	"github.com/uber/ringpop-go/shared"
 	"github.com/uber/ringpop-go/swim"
 	"github.com/uber/tchannel-go"
 )
 
-// Options to create a Ringpop with
-type Options struct {
-	Logger        log.Logger
-	Statter       log.StatsReporter
-	ReplicaPoints int
-}
-
-func defaultOptions() *Options {
-	logger := log.NewLoggerFromLogrus(&logrus.Logger{
-		Out: ioutil.Discard,
-	})
-
-	opts := &Options{
-		Logger:        logger,
-		Statter:       new(noopStatsReporter),
-		ReplicaPoints: 100,
-	}
-
-	return opts
-}
-
-func mergeDefault(opts *Options) *Options {
-	def := defaultOptions()
-
-	if opts == nil {
-		return def
-	}
-
-	if opts.Logger == nil {
-		opts.Logger = def.Logger
-	}
-
-	if opts.Statter == nil {
-		opts.Statter = def.Statter
-	}
-
-	if opts.ReplicaPoints <= 0 {
-		opts.ReplicaPoints = def.ReplicaPoints
-	}
-
-	return opts
-}
-
 // Ringpop is a consistent hash-ring that uses a gossip prtocol to disseminate changes around the
 // ring
 type Ringpop struct {
-	app     string
-	address string
+	config *Configuration
+
+	identityResolver IdentityResolver
```

Comment:
Is it necessary to hang onto the resolver like this? Why choose this way over resolving identity and hanging onto that instead?